### ISSUE TITLE
10.0 | New ProgressManager Extension

### ISSUE THREAD

**Issue by llaville** [overtrue/phplint#222]
The new ProgressManager extension will be in charge to handle properly all kinds all existing progress bars.

That's includes : 

- current https://github.com/overtrue/phplint/blob/9.6/src/Extension/ProgressBar.php component
- current https://github.com/overtrue/phplint/blob/9.6/src/Extension/ProgressIndicator.php
- current https://github.com/overtrue/phplint/blob/9.6/src/Extension/ProgressPrinter.php

Depends on new [Extension Architecture](https://github.com/overtrue/phplint/issues/221) that should be implemented first before to migrate current 3 extensions

**Comment by llaville** [comment]
## Flags behavior for the new extension

```text
  -p, --progress[=PROGRESS]          Set type of progress output (auto, quiet, plain, dots) [default: "auto"]
      --no-progress                  Suppress the progress output (same as --progress quiet)
```

## Keep BC with previous 9.x versions 

- with the `--no-progress` flag. This is equivalent now (alias) to `--progress quiet`.
- when `-vv` (SF very verbose mode (level 2+) is On). This is equivalent now to `--progress plain`.
- with `--progress printer`. This is equivalent now to `--progress dots` or `--progress auto` (and even `--progress` without specify `auto` value by default)

## With unknown `--progress` value

No errors/exceptions are raised; This is equivalent to `--progress quiet` (without progress widget) !

I.e:  `--progress foo` run like `--progress quiet`

Inspired by https://docs.docker.com/reference/cli/docker/buildx/build/#progress

**Comment by llaville** [comment]
Initially planned for major version 10.0; Will be finally integrated into version 9.8.0

See commit https://github.com/overtrue/phplint/commit/608b072a14abb09df3d745f8cc6c8c2a7c0fa753

**Comment by llaville** [comment]
Code of branch `9.8` is currently under rewrites, especially for Plugin/Extension Architecture.

So I've re-opened this report to annoted changes that will be commit soon.

**Comment by llaville** [comment]
Recent code refactoring by commit 2f0dc770ca9ed40a909f31fcbbe439649dd88650 have introduced following changes : 

- wrapper around Symfony/Console `Terminal` was definitively removed; we use now the default class provided by Symfony Console Component.
- the PHPLint specialized `EventDispatcher` is marked as deprecated, and will be removed in next major release. We use now the default/standard Symfony EventDispatcher
- all Progress Widgets (`bar`, `printer`, `indicator`) did not implement anymore the `Symfony\Component\EventDispatcher\EventSubscriberInterface` contract (useless)
- the `ProgressManager` used now the new `Overtrue\PHPLint\Helper\ProcessHelper`. Code was extracted from the old version of `Overtrue\PHPLint\Output\ConsoleOutput`
-  New version of `Overtrue\PHPLint\Output\ConsoleOutput` extends directly the Symfony `Symfony\Component\Console\Output\ConsoleOutput` class

> [!CAUTION]
> Coulde be considered as a BC break, because API was public, but I've definitively removed methods from `Overtrue\PHPLint\Output\ConsoleOutputInterface` contract that are now implemented in a dedicated `Helper` (`ProgressHelper`)

**Comment by llaville** [comment]
PHPLint 9.8 is almost ready for it's first Release Candidate:  Read more at https://github.com/overtrue/phplint/issues/239#issuecomment-5155378374 for final ROADMAP

**Comment by llaville** [comment]
Features is completed for the version 9.8.0-RC1
